feat(LinstorSR): use a specific resource group for DB and HA

Wescoeur opened the pull request.

Wescoeur committed:
feat(LinstorSR): use a specific resource group for DB and HA

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>

Nambrok commented on drivers/linstorvolumemanager.py:
```diff
@@ -1359,6 +1372,7 @@ def destroy(self):
 
             # 4.5. Destroy group and storage pools.
             self._destroy_resource_group(self._linstor, self._group_name)
+            self._destroy_resource_group(self._linstor, self._build_ha_goup_name(self._base_group_name))
```
It is weird having a call here to get the HA group name.

Nambrok commented on drivers/linstorvolumemanager.py:
```diff
@@ -1744,7 +1768,8 @@ def _create_sr(
-        pools = lin.storage_pool_list_raise(filter_by_stor_pools=[group_name])
+        storage_pool_name = group_name
+        pools = lin.storage_pool_list_raise(filter_by_stor_pools=[storage_pool_name])
         pools = pools.storage_pools
```
Could be done in one line to avoid changing `pools` type.

Nambrok commented on drivers/linstorvolumemanager.py:
```diff
@@ -1825,49 +1850,28 @@ def _create_sr(
+              lin,
+              ha_group_name,
+              storage_pool_name,
+              3,
```
Hardcoded value?

Wescoeur replied:
Yes. Fixed for each installation. No reason to use another value for the replication.

Nambrok commented on drivers/linstorvolumemanager.py:
```diff
@@ -2149,27 +2154,57 @@ def _get_storage_pools(self, force=False):
 
+        group_name = self._group_name
+        if high_availability:
+            group_name = self._build_ha_goup_name(self._base_group_name)
```
`self._build_ha_group_name()`

Nambrok commented on drivers/linstorvolumemanager.py:
```diff
@@ -2849,6 +2943,12 @@ def _build_group_name(cls, base_name):
+    # Used to store important data in a HA context,
+    # i.e. a replication count of 3.
+    @classmethod
+    def _build_ha_goup_name(cls, base_name):
```
`group`

Wescoeur committed:
feat(linstor-manager): add `getNodePreferredInterface` helper

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>

Wescoeur committed:
fix(linstorvolumemanager): blocks deletion of default network interface

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>

Wescoeur committed:
feat(linstorvolumemanager): change logic of `get_resources_info`:
- Add a nested level "nodes" for each resource
- Add a "uuid" attr on resources
- Rename LINSTOR "uuid" to "linstor-uuid"
- Optimize code

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>

Wescoeur committed:
feat(linstor-manager): add error codes to healthCheck helper

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>

Wescoeur committed:
fix(LinstorSR): fix bad exception reference during snapshot

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
